unblob/file_utils.py

```diff
@@ -16,6 +16,19 @@ class Endian(enum.Enum):
     BIG = ">"
 
 
+def bits(file: io.BufferedIOBase) -> Iterator[int]:
+    """bit-wise reading of file in little-endian mode"""
+    try:
+        cur_bytes = file.read(8)
+        while cur_bytes:
+            for b in cur_bytes:
+                for i in range(7, -1, -1):
+                    yield (b >> i) & 1
+            cur_bytes = file.read(8)
+    except EOFError:
```

Comment:
File read won't raise EOFError ever, just returns an empty string if reached the end of file